### PR TITLE
Enable dependency info dumping tests for abspath

### PR DESCRIPTION
**Pull request by Qining**
Enable tests like:
```
glslc -M /usr/local/a.vert
```
The expected result is:
```
a.vert.spv: /usr/local/a.vert <abspath of included files>
```

### CODE REVIEW

File: glslc/test/option_dash_M.py [hunk 1]
```diff
@@ -268,32 +268,37 @@ class TestDashCapMDashCapMT(DependencyInfoStdoutMatch):
...
 class TestDashCapMInputAbsolutePathWithInclude(DependencyInfoStdoutMatch):
     """Tests -M have included files represented in absolute paths when the input
     file is represented in absolute path.
     E.g. Assume a.vert has '#include "b.vert"'
          glslc -M /usr/local/a.vert
       => a.vert.spv: /usr/local/a.vert /usr/local/b.vert
     """
-    shader_a = FileShader('#version 140\n#include "b.vert"\nvoid main(){}\n',
-                          '.vert')
-    shader_b = FileShader('void foo(){}\n', '.vert')
```
Comment (Qining) [inline glslc/test/option_dash_M.py]: My original code is wrong, as `include "b.vert"` won't point to shader_b.

**Comment by dekimir** [comment]
Note that we don't support absolute paths in `#include` anymore.

**Comment by Qining** [comment]
Yes, and we don't have tests for `include "<abs path>"`. The enabled tests are only for:
"when input file is represented in absolute path, the dependent files should be represented in absolute paths too."

**Comment by antiagainst** [comment]
+1.

As discussed yesterday, it would be better to move all the test framework related stuff to their expected files. But that would be another pull request.

**Comment by dekimir** [comment]
+2

**Comment by dneto0** [comment]
Rebased and pushed into master as a18ea5bee5a

**Pull request closed without merging**